main.tf

```diff
@@ -12,6 +12,14 @@ locals {
       }
     ] : null
   }
+
+  format_env_args = {
+    for instance in var.instances : instance.name =>
+    length(lookup(instance, "environmentd_extra_args", [])) > 0 ? [
+      for item in instance.environmentd_extra_args :
+      "--system-parameter-default=${item.name}=${item.value}"
+    ] : null
+  }
 }
```

Comment:
I'm not sure about this:
not all values in `environmentd_extra_args` need to be system parameter defaults, it could be a non system parameter flag as well. 

I think we should either make users input a list[str] like 
```yaml
environmentd_extra_args:
  - --arg1=x
  - --arg2=y
```

or take in a dict of `system_parameter_defaults`  more specifically

Reply:
Oh yes, good point. Let me know what you think of the last change. Essentially it will be up to the user to specify this in the correct format similar to how we have it with the emulator.

```json
    environmentd_extra_args = [
      "--system-parameter-default=max_clusters=100",
      "--system-parameter-default=max_connections=100"
    ]
```